### PR TITLE
Added missing sponsor relationships

### PR DESCRIPTION
**Pull request by Naapperas**

### CODE REVIEW

**Review by pedrosilva17 — DISMISSED** [review]

**Review by CiscoPr — DISMISSED** [review]

File: app/Models/Edition.php [hunk 1]
```diff
@@ -39,4 +40,9 @@ public function quests(): HasMany
     {
         return $this->hasMany(Quest::class);
     }
+
+    public function sponsors(): BelongsToMany
+    {
+        return $this->belongsToMany(Sponsor::class);
+    }
```
Comment (ttoino) [inline app/Models/Edition.php]: Shouldn't this be a hasMany relationship?

Comment (ttoino) [reply app/Models/Edition.php]: Looking at the docs, belongsToMany is used in many to many relations, and, from what I understand, sponsors can only belong to one edition, right?

File: app/Models/Company.php [hunk 2]
```diff
@@ -31,4 +32,9 @@ public function user(): BelongsTo
     {
         return $this->belongsTo(User::class);
     }
+
+    public function sponsors(): BelongsToMany
+    {
+        return $this->belongsToMany(Sponsor::class);
+    }
```
Comment (ttoino) [inline app/Models/Company.php]: Same as the other comment.

File: app/Models/Edition.php (hunk 1 above)
Comment (Naapperas) [reply app/Models/Edition.php]: `Sponsor` is the association class representing the "Many to Many" association between `Company` and `Edition`. I gave a quick look at the documentation but honestly couldn't quite figure out what should be used here: perhaps you're right.

Comment (Naapperas) [reply app/Models/Edition.php]: But to answer your question, yes, sponsors only belong to a single edition and represent a single company

Comment (ttoino) [reply app/Models/Edition.php]: Laravel does not allow association classes like that, you're defining the relationship between companies and sponsors, so the correct one would be hasMany I think

Comment (Naapperas) [reply app/Models/Edition.php]: On it.

**Review by ttoino — APPROVED** [review]

**Review by diogofonte — APPROVED** [review]
Looks good!

**Review by pedrosilva17 — APPROVED** [review]
I was convinced `belongsToMany()` represented a one to many relationship. Thank you @ttoino 👍

**Pull request merged**